crux-dbg/src/Crux/Debug/LLVM.hs

```diff
@@ -150,9 +172,74 @@ do_breakpoint :: (wptr ~ CLE.ArchWidth arch)
               -> LCS.OverrideSim (SC.LLVMPersonality sym) sym (CLI.LLVM arch) r args ret ()
 do_breakpoint _gv _sym _ = return ()
 
+-- TODO: export in crux
+lookupString :: (LCB.IsSymInterface sym, CLM.HasLLVMAnn sym, CLO.ArchOk arch)
+             => LCS.GlobalVar CLM.Mem
+             -> LCS.RegEntry sym (CLT.LLVMPointerType (CLE.ArchWidth arch))
+             -> C.OverM personality sym (CLI.LLVM arch) String
+lookupString mvar ptr =
+  do sym <- LCS.getSymInterface
+     mem <- LCS.readGlobal mvar
+     bytes <- liftIO (CLM.loadString sym mem (regValue ptr) Nothing)
+     return (BS8.unpack (BS.pack bytes))
+
+-- TODO: export in crux
+withSolverAdapter :: CCS.SolverOffline -> (WS.SolverAdapter st -> a) -> a
+withSolverAdapter solverOff k =
+  case solverOff of
+    CCS.Boolector -> k WS.boolectorAdapter
+    CCS.DReal -> k WS.drealAdapter
+    CCS.SolverOnline CCS.CVC4 -> k WS.cvc4Adapter
+    CCS.SolverOnline CCS.STP -> k WS.stpAdapter
+    CCS.SolverOnline CCS.Yices -> k WS.yicesAdapter
+    CCS.SolverOnline CCS.Z3 -> k WS.z3Adapter
+
+do_debug_assert :: ( CLO.ArchOk arch
+                  , LCB.IsSymInterface sym
+                  , CLM.HasLLVMAnn sym
+                  , sym ~ (WEB.ExprBuilder t st fs)
+                  , SC.Architecture arch' t
+                  , SC.SymbolicArchitecture arch' t
+                  , ext ~ CLI.LLVM arch
+                  )
+                => C.CruxOptions
```

Comment:
Can we change this to compute the necessary solver configuration/adapter before we instantiate the override? Two reasons come to mind:
- It would be nice to not have this code depend on crux, since it doesn't really have any other dependency on crux
- I am pretty sure we are going to add some other similar overrides, so re-computing the necessary solver each time is a bit wasteful

Reply:
I moved the solver parsing code out.